humanprotocol/human-protocol
# [HUMAN App] Healthcheck endpoints

## Description
Adding `healthcheck` functionality to Human App Server

## Summary of changes
Added new `health` endpoints: `ping` and `check` with different purpose:
- `ping` is useful when you want to check that the server is up, running and can be reached via HTTP; it supplies some information about version as well, so you can easily understand what's is deployed w/o having access to CI/infra tools
- `check` - the main endpoint that provides information about operability and health of dependencies.

Before this PR, application crashed in case of connection errors to Redis and wasn't able to recover until Redis is green. In order to avoid that we have to add error handling to underlying client of `cache-manager-redis-store` instance with some basic configuration:
- `disableOfflineQueue: true` is necessary to not make incoming requests "wait" if they rely on Redis and just fail
- throttle error log because otherwise it floods with new message every 500ms
With this change application will keep serving routes that do not rely on Redis (e.g. profile, sign-in, etc.), which is believed to be better than full outage.

Also removed some unused packages.

## How test the changes
### To test `/health/ping`
1. `export GIT_HASH=$(git rev-parse HEAD) && yarn start`
2. Call the endpoint and check the body

### To test `/health/check`
- run the app with redis server running; call the endpoint and check the result, should be `ok`
- stop the redis server while app is running; call the endpoint and check the results, should be `error`

## Release notes
In order to enable commit hash for non-local envs we will need to include it to "start" command to include `GIT_SHA` from [RENDER_GIT_COMMIT](https://docs.render.com/environment-variables#:~:text=service%20or%20deploy.-,RENDER_GIT_COMMIT,-The%20commit%20SHA) env variable:
```
GIT_HASH=$RENDER_GIT_COMMIT yarn start:prod
```
or put it somehow via env variables config if possible

## Related issues
Closes #2646
